gemini/orchestration/llamaindex_workflows.ipynb

```diff
@@ -0,0 +1,1127 @@
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "0e101376082c"
+      },
+      "source": [
+        "### Get data"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "f15a704ea11b"
+      },
+      "outputs": [],
+      "source": [
+        "!mkdir -p './data'\n",
+        "!wget 'https://www.gutenberg.org/cache/epub/64317/pg64317.txt' -O 'data/gatsby.txt'"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "927a7ffd9ad8"
+      },
+      "source": [
+        "### Set credentials"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 4,
+      "metadata": {
+        "id": "7d11aac2947b"
+      },
+      "outputs": [],
+      "source": [
+        "import google.auth\n",
+        "import google.auth.transport.requests\n",
+        "\n",
+        "# credentials will now have an api token\n",
+        "credentials = google.auth.default(quota_project_id=PROJECT_ID)[0]\n",
+        "request = google.auth.transport.requests.Request()\n",
+        "credentials.refresh(request)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "af476af08250"
+      },
+      "source": [
+        "## Workflow"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "456efd13df2a"
+      },
+      "source": [
+        "### Set up the LLM"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "695792a24ba9"
+      },
+      "outputs": [],
+      "source": [
+        "safety_config = [\n",
+        "    SafetySetting(\n",
+        "        category=HarmCategory.HARM_CATEGORY_DANGEROUS_CONTENT,\n",
+        "        threshold=HarmBlockThreshold.BLOCK_ONLY_HIGH,\n",
+        "    ),\n",
+        "    SafetySetting(\n",
+        "        category=HarmCategory.HARM_CATEGORY_HARASSMENT,\n",
+        "        threshold=HarmBlockThreshold.BLOCK_ONLY_HIGH,\n",
+        "    ),\n",
+        "    SafetySetting(\n",
+        "        category=HarmCategory.HARM_CATEGORY_SEXUALLY_EXPLICIT,\n",
+        "        threshold=HarmBlockThreshold.BLOCK_ONLY_HIGH,\n",
+        "    ),\n",
+        "]\n",
+        "embedding_model = VertexTextEmbedding(\n",
+        "    model_name=\"text-embedding-004\", credentials=credentials\n",
+        ")\n",
+        "llm = Vertex(\n",
+        "    model=\"gemini-pro\",\n",
+        "    temperature=0.2,\n",
+        "    max_tokens=3000,\n",
+        "    safety_settings=safety_config,\n",
+        "    credentials=credentials,\n",
+        ")\n",
+        "\n",
+        "Settings.embed_model = embedding_model\n",
+        "Settings.llm = llm"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "1b99e7e4aa0d"
+      },
+      "source": [
+        "### Define Event classes\n",
```

Comment:
Why do you need to create custom Event classes? Should these be added to the LlamaIndex library directly?

Reply:
LlamaIndex only provides the StartEvent and StopEvent. I believe they want them to be completely user defined.
According to LlamaIndex:

"Events are user-defined pydantic objects. You control the attributes and any other auxiliary methods. In this case, our workflow relies on a single user-defined event, the JokeEvent."

"While events are use-defined, there are two special-case events, the StartEvent and the StopEvent"

https://docs.llamaindex.ai/en/stable/module_guides/workflow/

LMK if OK to resolve @holtskinner.

Reply:
Ok, sounds good!